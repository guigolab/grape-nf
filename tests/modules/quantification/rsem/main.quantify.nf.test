nextflow_process {

    name "Test Process quantify"
    script "modules/quantification/rsem/main.nf"
    process "quantify"

    tag "module"

    setup {
        run("index") {
            script "modules/quantification/rsem/main.nf"
            process {
                """
                input[0] = file("${baseDir}/data/genome.fa.gz")
                input[1] = file("${baseDir}/data/annotation.gtf.gz")
                """
            }
        }
    }

    test("Should quantify with RSEM") {

        when {
            process {
                """
                input[0] = index.out
                input[1] = Channel.from([
                    [ "sample3", "test3", file("${baseDir}/data/sample3_m4_n10_toTranscriptome.bam"), "bam", "TranscriptomeAlignments", true, "MATE2_SENSE" ]
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("Should quantify with RSEM (with credibility intervals)") {

        when {
            params {
                rsemCalcCI = true
            }
            process {
                """
                input[0] = index.out
                input[1] = Channel.from([
                    [ "sample3", "test3", file("${baseDir}/data/sample3_m4_n10_toTranscriptome.bam"), "bam", "TranscriptomeAlignments", true, "MATE2_SENSE" ]
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
