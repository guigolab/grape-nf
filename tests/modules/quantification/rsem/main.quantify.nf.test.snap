{
    "Should quantify with RSEM": {
        "content": [
            {
                "0": [
                    [
                        "sample3",
                        "test3",
                        "sample3.isoforms.results:md5,a2ddfd7d7919830d69f941176ddff86f",
                        "tsv",
                        "TranscriptQuantifications",
                        true,
                        "MATE2_SENSE"
                    ]
                ],
                "1": [
                    [
                        "sample3",
                        "test3",
                        "sample3.genes.results:md5,568855cd154369988f0e9120660dcea6",
                        "tsv",
                        "GeneQuantifications",
                        true,
                        "MATE2_SENSE"
                    ]
                ],
                "genes": [
                    [
                        "sample3",
                        "test3",
                        "sample3.genes.results:md5,568855cd154369988f0e9120660dcea6",
                        "tsv",
                        "GeneQuantifications",
                        true,
                        "MATE2_SENSE"
                    ]
                ],
                "isoforms": [
                    [
                        "sample3",
                        "test3",
                        "sample3.isoforms.results:md5,a2ddfd7d7919830d69f941176ddff86f",
                        "tsv",
                        "TranscriptQuantifications",
                        true,
                        "MATE2_SENSE"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-30T17:37:25.520162"
    }
}